### ISSUE TITLE
Error connect to wifi whose password has space

### ISSUE THREAD

**Issue by dainguyenSH** [SamsungARTIK/artik-sdk-js#6]
Hi everyone,

I have a issue when I try to connect artik to the wifi whose password has space. When I configed by connect() method, I got message "Bad arguments".

One more issue, if the SSID has any special character or Vietnamese character as (', ư, ê, ầ...), that character is convert to format UTF-8 ( "ư" => "\\xc6\\xb0",     ' => "\xe2\x80\x99" ) so I can not parse to Json by JSON.parse()

Please check these and help me. Thank you very much!

**Comment by glemercier** [comment]
What version of the ARTIK SDK are you using? We fixed a bug in this area for the recently deployed 1.8.2 version, please try with this one.

**Comment by dainguyenSH** [comment]
Hi **glemercier,**

Thank you for your feedback. I am using artik 530, I upgrade libartik-sdk and artik-sdk-js to 1.8.2 but still got message "Bad arguments" when the password has space. I had reported a bug before which the SSID has space. And this times, it's PASSWORD. Please check again and check more about special characters that I noted above. Thank you very much!

**Comment by glemercier** [comment]
About the spaces in the password, I wrote  a patch to fix this issue in the C layer: https://github.com/SamsungARTIK/artik-sdk/commit/e82c58ed8e98f3421e87e961903ca4c01938c159

If you want to use this patch you need to rebuild the ARTIK SDK C libraries using the tip of the **1.8-aintenance** branch or wait for the next release of the ARTIK SDK.

About the unicode issue in JSON, I'm looking into it.

**Comment by glemercier** [comment]
I tried with pinyin characters and I cannot reproduce your unicode issue. I created a hostpot on my phone with pinyin characters and modified the **wifi-example.js** file to try and connect to it. When I run the example program it works:

```
var ssid = '啊test啊';
var pwd = 'aa bb cc dd';
```

```
[root@artik ~]# node node_modules/artik-sdk/examples/wifi-example.js
Current Wifi Information:       {
                "connected": false,
                "error": false,
                "name": "null",
                "bssid": "null",
                "frequency": 0,
                "signal": 0,
                "encryption": "null",
                "mode": "null"
        }

[ { name: '啊test啊',
    bssid: 'aa:bb:cc:dd:ee:ff',
    frequency: 2412,
    signal: -29,
    encryption: 'WPA2,',
    mode: 'none' } ]
Found SSID 啊test啊, connecting...
connected
```
Can you elaborate on how you reproduce the issue on your end?

**Comment by dainguyenSH** [comment]
Please modify your SSID is "test's" or "test thư" which "ư" is a Vietnamese character.

**Comment by glemercier** [comment]
Works as well on my end:

```
var ssid = 'testư';
var pwd = 'aa bb cc dd';
```

```
[root@artik ~]# node node_modules/artik-sdk/examples/wifi-example.js
Current Wifi Information:       {
                "connected": false,
                "error": false,
                "name": "null",
                "bssid": "null",
                "frequency": 0,
                "signal": 0,
                "encryption": "null",
                "mode": "null"
        }
[ {  name: 'testư',
    bssid: 'aa:bb:cc:dd:ee:ff',
    frequency: 2462,
    signal: -23,
    encryption: 'WPA2,',
    mode: 'none' } ]
Found SSID testư, connecting...
connected
```

**Comment by dainguyenSH** [comment]
Oh my God! This bug is not exist now but it appeared before

**Comment by glemercier** [comment]
It was probably fixed in latest 1.8.2 version, see patch here: https://github.com/SamsungARTIK/artik-sdk/commit/79db4dc86089fda4ba5e02e88bae532f71c33650.
